### PR TITLE
add null-safety checks in ValidateController.ProcessResource()

### LINKED ISSUE #5627: fix: NullReferenceException in ValidateController.ProcessResource() when Parameters lacks resource param
## Problem

`ValidateController.ProcessResource()` at line 74 has the same null-dereference pattern as #5114 (`ParameterCompatibleFilter.ParseResource()`):

```csharp
resource = parameterResource.Parameter.Find(param => param.Name.Equals("resource", StringComparison.OrdinalIgnoreCase)).Resource;
```

If `.Find()` returns `null` (no "resource" parameter exists), accessing `.Resource` throws `NullReferenceException`.

Additionally, line 60 does not null-check `parameterResource.Parameter`:
```csharp
var profileFromParameters = parameterResource.Parameter.Find(param => ...);
```

## Root Cause

No null-safety on the result of `.Find()` or on the `Parameter` collection itself. The Copilot PR #5115 fixes this in `ParameterCompatibleFilter.cs` but the same pattern exists in `ValidateController.ProcessResource()`.

## Proposed Fix

Add null-safety checks matching the pattern from PR #5115:
- Null-check `parameterResource.Parameter` before calling `.Find()`
- Null-check the `.Find()` result before accessing `.Resource`

## Affected File
- `src/Microsoft.Health.Fhir.Shared.Api/Controllers/ValidateController.cs` lines 60, 74

### PR DESCRIPTION
**Pull request by v-rachitsh**
Fixes #5627

## Related issues
[AB#198060](https://dev.azure.com/microsofthealth/Health/_workitems/edit/198060)

## Summary

Adds null-safety checks in `ValidateController.ProcessResource()` to prevent `NullReferenceException` when a Parameters resource is missing the expected "resource" parameter or has a null Parameter collection.

This is the same class of bug fixed in PR #5115 for `ParameterCompatibleFilter.ParseResource()`, but in a separate code path.

## Root Cause

`ProcessResource()` at line 74 called `.Find(...).Resource` without null-checking the result of `.Find()`. If no parameter named "resource" exists, `.Find()` returns `null` and accessing `.Resource` throws `NullReferenceException`.

Additionally, `parameterResource.Parameter` was accessed without null-checking, which would also throw if the Parameter collection itself is null.

### `ValidateController.cs`
- Added `?.` null-conditional operator on `resource.TypeName` check
- Added `?.` null-conditional operator on `parameterResource.Parameter` access
- Split the `.Find(...).Resource` chain into separate null-safe steps matching the pattern from PR #5115

### `ValidateControllerTests.cs`
- Added 3 new test cases:
  - `GivenParametersWithNullParameterCollection_WhenValidating_ThenShouldNotThrowNullReferenceException`
  - `GivenParametersWithMissingResourceParameter_WhenValidating_ThenShouldNotThrowNullReferenceException`
  - `GivenParametersWithNullResourceParameter_WhenValidating_ThenShouldNotThrowNullReferenceException`

## Verification

Structural verification: the fix follows the exact same null-safety pattern established in PR #5115 for `ParameterCompatibleFilter.cs`. Each `.Find()` result is now checked for null before accessing `.Resource`.

### Files Changed
- src/Microsoft.Health.Fhir.Shared.Api/Controllers/ValidateController.cs — Added null-conditional operators and null-safe Find result handling
- src/Microsoft.Health.Fhir.Shared.Api.UnitTests/Controllers/ValidateControllerTests.cs — Added 3 regression tests for null Parameters edge cases

### Verification
- Fix follows identical pattern from PR #5115 (ParameterCompatibleFilter)
- 3 new unit tests cover all null-dereference paths

### CODE REVIEW

File: src/Microsoft.Health.Fhir.Shared.Api/Controllers/ValidateController.cs [hunk 1]
```diff
@@ -45,19 +45,23 @@ public ValidateController(IMediator mediator, ResourceDeserializer resourceDeser
...
+            Resource originalResource = resource;
+
             ProcessResource(ref resource, ref profile);
 
             Uri profileUri = GetProfile(profile);
 
-            return await RunValidationAsync(resource.ToResourceElement(), profileUri);
+            // This endpoint has no id to fall back to, so if the Parameters payload did not
+            // carry an actual resource, validate the original payload instead of throwing.
+            return await RunValidationAsync((resource ?? originalResource).ToResourceElement(), profileUri);
```
Comment (LTA-Thinking) [inline src/Microsoft.Health.Fhir.Shared.Api/Controllers/ValidateController.cs]: This doesn't make sense for this operation. If the inner resource is null there is nothing to validate, it should return a bad request.

Comment (v-rachitsh) [reply src/Microsoft.Health.Fhir.Shared.Api/Controllers/ValidateController.cs]: Updated the code.

**Review by LTA-Thinking — APPROVED** [review]
